### DISCUSSION TITLE
ImGui not linking, can't even IMGUI_CHECKVERSION();

### DISCUSSION THREAD

**Discussion by carlosdr02** [ocornut/imgui#4584] [New Users Build/Link/Run issues ONLY!]
I've just downloaded ImGui and included it in my Visual Studio project by Properties > C/C++ > General > Additional Include Directories, and the header files are added correctly, but seems like .cpp file aren't. I get the following error when trying to `IMGUI_CHECKVERSION();`:

`application.obj : error LNK2019: unresolved external symbol "bool __cdecl ImGui::DebugCheckVersionAndDataLayout(char const *,unsigned __int64,unsigned __int64,unsigned __int64,unsigned __int64,unsigned __int64,unsigned __int64)" (?DebugCheckVersionAndDataLayout@ImGui@@YA_NPEBD_K11111@Z) referenced in function "public: __cdecl Application::Application(void)" (??0Application@@QEAA@XZ)`

It's about the function `ImGui::DebugCheckVersionAndDataLayout()`, which is what `IMGUI_CHECKVERSION()` calls. 

I have tried other ways of including the library, like just dragging the files into the folder where all my source code is, still not working. Any ideas?

Thx.

**Comment by ocornut** [comment]
﻿Dear ImGui is not linking because you aren’t linking it, it is as simple as that and you’ll understand at this point this is a question about how to use your build system rather than a question about dear imgui.

**In Visual Studio you need to add all sources files to your project source files.**

**Reply by ocornut** [reply]
I recently wrote a new guide which hopefully can clarify this for some users:
  https://github.com/ocornut/imgui/wiki/Getting-Started

Sorry for the dry answer but as maintainers the amount of general, non-project specific, C++ building question gets overwhelming. Any C/C++ projects gets those questions but when you factor in the amount of backends/technology it becomes more!.
